parser/src/semantic/04_check_dont_return_secret.ts

```diff
@@ -22,6 +22,16 @@ export class CheckDontReturnSecretVisitor extends Visitor {
       this.visit(node.returnType);
       this.path.pop();
       this.isInReturn = false;
+
+      for (const annotation of node.annotations) {
+        if (annotation instanceof RestAnnotation) {
+          for (const arg of node.fieldsAndSpreads) {
+            if (annotation.method === "GET" && (arg as Field).secret) {
```

Comment:
Isso aqui é um problema. Esse check está acontecendo antes do `ExpandSpreadsVisitor` (05), então aqui os spreads ainda não foram expandidos e essa análise pode falhar. Por exemplo:

```
type Foo {
  a: int !secret
}

@rest GET /{a}/{b}
fn getSomething(b: string, ...Foo): json
```

Nesse caso precisa dar erro, mas isso só é possível depois da análise 5. Bom incluir esse exemplo no spec, por sinal.

Solução: troque esse for loop para usar `node.args`. E inverta a ordem do visitor 4 e 5. Requer mudar o nome dos arquivos e alterar o `analyser.ts`.